### PR TITLE
ClientsGrid component

### PR DESCRIPTION
**Pull request by pichfl**
Since we use the clients grid on multiple pages, I thought it could be useful to abstract the grid into a component.

Extends #361

### CODE REVIEW

**Comment by marcoow** [comment]
I don't think we always want to show all clients though?

**Comment by marcoow** [comment]
I'm also not sure Ah sorry, just saw there's an option for including and excluding. I'm not sure we actually want to show the description on all pages though? Maybe it makes more sense extracting only each grid item so that we can reassemble those?

Also the shuffling will lead to problems with SSR as the order will be different in the SSR response than what is calculated when the app rehydrates in the browser so that the DOM changes on rehydration which looks strange to the user.

**Comment by pichfl** [comment]
As of now all pages required the descriptions. We could make them optional as well. I just thought it might be sensible to have the content separated from the actual html as it’s very repetitive across pages.

Re shuffling: I feared as much and we can easily drop that option but I thought it might be useful.

File: src/ui/components/ClientsGrid/component.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    if (included.length > 0) {
+      clients = clients.filter(client => included.includes(client.id));
+    }
+
+    if (excluded.length > 0) {
+      clients = clients.filter(client => !excluded.includes(client.id));
+    }
+
+    if (this.args.shuffle) {
+      shuffle(clients);
```
Comment (marcoow) [inline src/ui/components/ClientsGrid/component.ts]: As discussed, we can't actually shuffle due to different renderings then in SSR and when the app rehydrates in the client.

File: src/ui/components/ClientsGrid/component.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    if (this.args.limit) {
+      return clients.slice(0, parseInt(this.args.limit, 10));
+    }
+
+    return clients;
+  }
+}
```
Comment (marcoow) [inline src/ui/components/ClientsGrid/component.ts:111]: I think this support too many options (which aren't used anyway). I'd rename `include` to `only` and drop support for `exclude` and `limit`. I don't think we'll need those.

**Review by marcoow — COMMENTED** [review]
Cool, this makes sense to me but let's simplify a bit.

**Review by marcoow — APPROVED** [review]

**Pull request merged**